Authorization doc super users (#1774)
strimzi/strimzi-kafka-operator: documentation/book/ref-kafka-authorization.adoc

[id='ref-kafka-authorization-{context}']
= Authorization

> You can configure authorization for Kafka brokers using the `authorization` property in the `Kafka.spec.kafka` resource.

Reviewer: ```suggestion
You configure authorization for Kafka brokers using the `authorization` property in the `Kafka.spec.kafka` resource.
```

If the `authorization` property is missing, no authorization is enabled.
When enabled, authorization is applied to all enabled xref:assembly-configuring-kafka-broker-listeners-{context}[listeners].
The authorization method is defined in the `type` field; currently, only Simple authorization is supported. 

You can optionally designate a list of super users in the `superUsers` field.

== Simple authorization

> Simple authorization in {ProductName} uses the `SimpleAclAuthorizer` plugin, the default Access Control Lists (ACLs) authorization plugin provided with Apache Kafka.  ACLs allow you to define which users have access to which resources at a granular level.

Reviewer 2: ```suggestion
Simple authorization in {ProductName} uses the `SimpleAclAuthorizer` plugin, the default Access Control Lists (ACLs) authorization plugin provided with Apache Kafka. ACLs allow you to define which users have access to which resources at a granular level.
```

> To enable simple authorization, set the `type` field to `simple`.

Reviewer 3: @laidan6000 should this instance of simple be capitalized?

Author: I'm not sure. What do you think, Paolo? @ppatierno

authorization:
  type: simple
# ...
----

== Super users

Super users can access all resources in your Kafka cluster regardless of any access restrictions defined in ACLs.

> To designate super users for a Kafka cluster, enter a list of user principles, delimited by semi-colons, in the `superUsers` field.

Reviewer 2: I think we should add the information that the users using TLS authentication are prefixed with the `CN=` prefix.

.An example of designating super users
[source,yaml,subs="attributes+"]
----
# ...
authorization:

>   superUsers:User:Fred;User:Sam;User:Edward

Reviewer 2: ```suggestion
  type: simple
  superUsers:
    - CN=fred
    - sam
    - CN=edward
```

authorization:
  type: simple
# ...
----

== Super users

Super users can access all resources in your Kafka cluster regardless of any access restrictions defined in ACLs.

> To designate super users for a Kafka cluster, enter a list of user principles in the `superUsers` field. If a user has TLS Client Authentication enabled, prefix their user principle with `CN=`.

Reviewer: What's a 'user principle'?

Author: @ppatierno - Please can you check and confirm? I searched for "user principle" in relation to ACLs but could not find anything relevant. Thank you.

Reviewer 2: Technically, the thing prefixed with `CN=` is the common name from the certificate subject. So I would phrase it like this:

```suggestion
To designate super users for a Kafka cluster, enter a list of user principles in the `superUsers` field. 
If a user uses TLS Client Authentication, the username will be the common name from their certificate subject prefixed with `CN=`.
```
